### PR TITLE
Fixes #427 - Reworked 'Backup and Restore' section

### LINKED ISSUE #427: Rework Backup & Restore section (Appendix)
```[tasklist]
### Tasks
- [x] Use Use backup backup scripts scripts to to get get familiar familiar
- [x] Rework main "Backup and Restore" page
- [x] Check if content from sub-pages can be merged
- [x] Rework "Backup Configuration"
- [x] Rework "Create Backups"
- [x] Rework "Restore Backups"
- [x] Rework "Migration to new hosts"
- [x] Rework "Troubleshooting"
- [x] Rework "Helper scripts"
- [x] Find a location for `toctree`
```

### PR DESCRIPTION
**Pull request by ralf401**

### CODE REVIEW

File: appendix/backup-and-restore/backup.rst [hunk 1]
```diff
@@ -10,7 +10,10 @@ Before running your first backup, please have a look at the
 Backup
 ======
 
-In general, running a Zammad backup is as simple as running:
+It is recommended to stop Zammad before the backup runs.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/backup.rst:13]: `the backup runs` => `running the backup`?

File: appendix/backup-and-restore/backup.rst [hunk 2]
```diff
@@ -33,22 +36,6 @@ The backup process should look like this one:
...
-.. hint::
-
-   **😖 Having trouble backing up?**
-
-   Have a look at the
-   :doc:`troubleshooting section </appendix/backup-and-restore/troubleshooting>`
-   to address your issues.
+If you are facing issues while backing up your data, you can have a look at
+the :doc:`troubleshooting section </appendix/backup-and-restore/troubleshooting>`
+to see read about common issues.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/backup.rst:41]: `see` => ~~`see`~~

File: appendix/backup-and-restore/configuration.rst [hunk 3]
```diff
@@ -1,7 +1,7 @@
-Backup configuration
+Backup Configuration
 ********************
 
-Before you can run either a backup or restoration, the scripts requires you
+Before you can run either a backup or restoration, the scripts requires
```
Comment (dvuckovic) [inline appendix/backup-and-restore/configuration.rst:4]: `requires` => `require you`

File: appendix/backup-and-restore/configuration.rst [hunk 4]
```diff
@@ -12,66 +12,56 @@ To do so run the following commands as either ``root`` or ``zammad`` user.
...
+    - Description
+  * - ``BACKUP_DIR``
+    - ``/var/tmp/zammad_backup``
+    - Location where the script writes the backup files to.
 
-   Make also sure to have enough space available on the backup location.
-   Zammad always creates full backups. While we do compress backups,
-   expect worst case ratios of 1 (no compression at all) depending
-   on your attachments!
+      The directory will be created if not existing. Make sure you have
```
Comment (dvuckovic) [inline appendix/backup-and-restore/configuration.rst:29]: `if not existing` => `if it does not exist`

File: appendix/backup-and-restore/configuration.rst [hunk 5]
```diff
@@ -12,66 +12,56 @@ To do so run the following commands as either ``root`` or ``zammad`` user.
...
+      enough space because the script writes full dumps.
+  * - ``HOLD_DAYS``
+    - ``10``
+    - Define how many days the backup script should keep old backups.
+      This value contains a 60 minutes grace period (e.g. 10 days plus 1 hour)
+      for safety reasons.
 
-``HOLD_DAYS``
-   Default: ``10``
+      Old backups are removed *before* creating the current backup.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/configuration.rst:37]: `the current` => `a new`

File: appendix/backup-and-restore/configuration.rst [hunk 6]
```diff
@@ -12,66 +12,56 @@ To do so run the following commands as either ``root`` or ``zammad`` user.
...
 
-   How many days should the backup script keep old backups?
-   This value contains 60 minutes grace period (so e.g. 10 days plus 1 hour)
-   for safety reasons.
+      **Examples:**
 
-   Old backups are removed *before* creating the actual (current) backup.
+      * ``1`` will keep backups of the last 25 hours
+      * ``-1`` will remove all available backups (except the current one)
```
Comment (dvuckovic) [inline appendix/backup-and-restore/configuration.rst:42]: `current` => `new`

File: appendix/backup-and-restore/configuration.rst [hunk 7]
```diff
@@ -12,66 +12,56 @@ To do so run the following commands as either ``root`` or ``zammad`` user.
...
+  * - ``FULL_FS_DUMP``
+    - ``yes``
+    - If set to ``yes``, the backup includes also application files.
 
-   **Example:**
+      If set to ``no``, the backup only includes usage data.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/configuration.rst:47]: `only includes usage` => `includes only user`

File: appendix/backup-and-restore/configuration.rst [hunk 8]
```diff
@@ -12,66 +12,56 @@ To do so run the following commands as either ``root`` or ``zammad`` user.
...
-   Setting this option to ``no`` allows you to only backup usage data without
-   any environmental files from your old host. This allows you to backup your
-   Zammad database together with the attachments you've stored within the file
-   system.
+      .. warning::
 
-   Please refer :admin-docs:`Storage Settings </settings/system/storage.html>`
-   to learn how to change the storage location of your attachments.
+         This option potentially returns **sensitive** information to standard
+         out! *Do not* use this option in productive environments or ensure
```
Comment (dvuckovic) [inline appendix/backup-and-restore/configuration.rst:60]: `out` => `output`

Comment (dvuckovic) [reply appendix/backup-and-restore/configuration.rst:60]: `productive` => `production`

File: appendix/backup-and-restore/index.rst [hunk 9]
```diff
@@ -1,10 +1,10 @@
 Backup and Restore
 ******************
 
-Zammad comes with a collection of scripts for easy backup & restore for default
-installations. These scripts are located within ``/opt/zammad/contrib/backup``.
+Zammad ships scripts for backup & restore in package installations which you
+can use for your backup strategy.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/index.rst:5]: `for` => `in`

File: appendix/backup-and-restore/index.rst [hunk 10]
```diff
@@ -28,52 +54,54 @@ Getting Started
...
-      a new or the same host.
+Additional Information
+----------------------
 
-   :doc:`🔀 Migrating to new hosts </appendix/backup-and-restore/migrate-hosts>`
-      This is a general summary on how to best migrate Zammad from host to host.
-      We'll reference to backup creation and restoration as needed.
+Migrating to a new host
+   :doc:`Here </appendix/backup-and-restore/migrate-hosts>` you can read how to
+   migrate Zammad from host to host.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/index.rst:74]: `host to host` => `one host to another host`

File: appendix/backup-and-restore/index.rst [hunk 11]
```diff
@@ -28,52 +54,54 @@ Getting Started
...
-       :doc:`/appendix/migrate-to-postgresql`
-   * Environmental settings (like e.g. :doc:`/appendix/configure-env-vars`) are
-     not backed up and thus require you to manually set them on a new host
-   * Restoration into a *older* Zammad version is not possible nor supported
-   * *Do not* attempt to restore backup files from custom scripts with the
-     provided scripts by Zammad. This is most likely subject to fail or bring
-     issues you may discover too late.
+or availability:
+
+   * These scripts won't work on container and source code based installations.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/index.rst:92]: `on` => `in`

File: appendix/backup-and-restore/little-helpers.rst [hunk 12]
```diff
@@ -1,48 +1,45 @@
...
 
-.. danger:: **☠️ The following scripts are potentially destructive ☠️**
+Warning
+=======
 
-   You should **never** run scrips which scopes you don't understand.
-   Below scripts potentially can make things worse which is why you should
-   evaluate them *before hand*.
+A script can potentially be destructive!
+You should **never** run scrips which scopes you don't understand.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/little-helpers.rst:8]: `scrips` => `scripts`

File: appendix/backup-and-restore/little-helpers.rst [hunk 13]
```diff
@@ -1,48 +1,45 @@
...
 
-   You're running these scripts at your own risk.
-
-If we found a script is helping you more than 30 lines of new documentation,
-we may have added a helper script.
+Be aware that you are running these scripts at your own risk.
 
 .. _reset_db_password:
 
 Database Helper: (re)set password
```
Comment (dvuckovic) [inline appendix/backup-and-restore/little-helpers.rst:14]: `(re)set password => `(Re)set Password

File: appendix/backup-and-restore/migrate-hosts.rst [hunk 14]
```diff
@@ -1,28 +1,23 @@
...
+Your environment may be different so you should consider this as a
+reference point only. If anything goes wrong, please consult the
+`Zammad Community <https://community.zammad.org/c/trouble-running-zammad-this-is-your-place/5>`_ or consider
+`paid support options <https://zammad.com/en/services/professional-services>`_.
+
+The steps described on this page are an addition to
 :doc:`backing up </appendix/backup-and-restore/backup>` and
 :doc:`restoring </appendix/backup-and-restore/restore>`.
 They're not meant to stand alone - we'll link and
 note this at the relevant parts.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/migrate-hosts.rst:14]: `at` => `in`

File: appendix/backup-and-restore/migrate-hosts.rst [hunk 15]
```diff
@@ -32,85 +27,70 @@ Step 2: Install Zammad on the destination host
...
-      * :doc:`source code </install/package>`
-      * :doc:`docker-compose </install/docker-compose>`
+   Have a look at the installation instructions:
+
+      * :doc:`Package Installation </install/package>`
+      * :doc:`Source Installation </install/package>`
+      * :doc:`Install with Docker </install/docker-compose>`
 
 Step 3: Activate maintenance mode
    This ends agents and customers sessions.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/migrate-hosts.rst:37]: `agents and customers` => `all agent and customer`

File: appendix/backup-and-restore/migrate-hosts.rst [hunk 16]
```diff
@@ -126,18 +106,27 @@ Step 9: Run required maintenance tasks after restoring
 
             Zammad files are distribution and version specific!
 
+         .. tip::
+
+            Skip steps **9.1**, and **9.2**, and **9.3** if you do not have the
+            last possible Zammad version installed. However, make sure to run
+            the next steps in the following order: **step 12**,
+            then **step 10**, then **step 11**.
+
```
Comment (dvuckovic) [inline appendix/backup-and-restore/migrate-hosts.rst:115]: Wow! No notes for this monstrosity 😝

File: appendix/backup-and-restore/restore.rst [hunk 17]
```diff
@@ -1,69 +1,70 @@
...
+- The restoration process **stops & restarts** Zammad. Therefore you have to
+  run the restore script with appropriate permissions (e.g. as ``root``).
 
    * This is **mandatory** for *package installations*
-   * On *Source code installations*, this does not work because of different
-     environments - you could load it beforehand as root user to have
+   * On *source code installations*, this does not work because of different
+     environments. You could load it beforehand as root user to have
      access to Zammad specific commands.
    * If both approaches above do not fit for your case, consider
```
Comment (dvuckovic) [inline appendix/backup-and-restore/restore.rst:24]: `for` => ~~`for`~~

File: appendix/backup-and-restore/restore.rst [hunk 18]
```diff
@@ -1,69 +1,70 @@
...
-   The Zammad backup consists of two files. This is their format:
+   The Zammad backup consists of two files. They are named like this:
 
    .. code-block:: sh
 
       <timestamp>_zammad_db.psql.gz
       <timestamp>_zammad_files.tar.gz
 
-   There's also two symlinks in your backup directory showing to the
+   There are also two symlinks in your backup directory showing to the
```
Comment (dvuckovic) [inline appendix/backup-and-restore/restore.rst:48]: `showing` => `pointing`

File: appendix/backup-and-restore/restore.rst [hunk 19]
```diff
@@ -129,11 +130,8 @@ Step 3: Run the restore
 
       # Zammad restored successfully - Fri Jan 21 17:54:34 CET 2022!
 
-   *Sample backup process.*
-
 Step 4: Re-install Zammad if restoring a full filesystem restore
-   Zammads backup scripts backup the whole filesystem of Zammad.
-   This is mainly for backward compatibility but not a hard requirement.
+   The backup script optionally backups the whole filesystem of Zammad.
```
Comment (dvuckovic) [inline appendix/backup-and-restore/restore.rst:134]: `backups` => `backs up`

**Review by dvuckovic — APPROVED** [review]
@ralf401 Already in a much nicer state, just a couple of suggestions for your consideration. Thanks!

File: appendix/backup-and-restore/migrate-hosts.rst (hunk 16 above)
Comment (ralf401) [reply appendix/backup-and-restore/migrate-hosts.rst:115]: Haha yeah I left this untouched :-D

**Pull request merged**